node-fs-writefile/cp.js

```diff
@@ -0,0 +1,20 @@
+import { writeFile, readFile } from 'node:fs/promises';
+
+const [, , origFile, newFile] = process.argv;
+
+if (!origFile.endsWith('.txt')) {
+  console.error('Source file must be .txt format');
+  process.exit(1);
+}
+if (!newFile.endsWith('.txt')) {
+  console.error('Destination file must be .txt format');
+  process.exit(1);
+}
+
+readFile(origFile, 'utf8')
+  .then((value) => {
+    writeFile(newFile, value)
+      .then((value) => value)
+      .catch((writeErr) => console.error(writeErr));
+  })
+  .catch((readErr) => console.error(readErr));
```

Comment:
It turns out that when you chain promises, the errors roll up to the outermost .catch handler. So you could write this as:
```
readFile(...)
  .then((value) => writeFile(newFile, value))
  .catch((err) => console.error(err));
```
This works because the `.then` handler returns a promise (the one returned by `writeFile`.